src/nvim/api/vim.c

```diff
@@ -140,12 +140,16 @@ String nvim_exec(String src, Boolean output, Error *err)
 ///
 /// @param command  Ex-command string
 /// @param[out] err Error details (Vim error), if any
-void nvim_command(String command, Error *err)
+void nvim_command(uint64_t channel_id, String command, Error *err)
   FUNC_API_SINCE(1)
 {
   try_start();
   do_cmdline_cmd(command.data);
   try_end(err);
+
+  if (channel_id == LUA_INTERNAL_CALL) {
```

Comment:
Wouldn't it make more sense to have something happen in `:packadd`'s code rather than every command?

Reply:
Oh.... that's pretty smart :) Probably just always call it in `:packadd`, regardless of whether it's this channel or not, right?

Reply:
It depends on what we're trying to do. :) `:packadd` is modifying `'rtp'`, yes, but why isn't _any_ change to `'rtp'` updating the related Lua variables?  Why is `:packadd` relevant?

Reply:
Ah, what happens is:

if you do `vim.cmd [[ packadd <package> ]]` in lua, the lua package path doesn't update until the next time you enter lua. This is not good, since then you can't require things that you've added.

I suppose though that you're right, we could instead go down to something that changes rtp and just do the fix there, rather than at the nlua_enter stage... I will look into doing it for the runtimepath generally, not just for packadd.